playwright/e2eTests/api-delegering.spec.ts

```diff
@@ -1,100 +1,228 @@
+      { name: standardApiDetails.name, department: standardApiDetails.department },
+      { orgNumber: standardOrgUser.orgNumber, reportee: standardOrgUser.reportee },
     );
   });
+
+  test('Delegate api to an organization by selecting from API providers filter', async ({
+    login,
+    apiDelegations,
+    page,
+  }) => {
+    //Login and cleanup state before running test
+    await login.loginWithUser(standardDaglUser.id);
+    await login.chooseReportee(standardDaglUser.reportee);
+    await apiDelegations.deleteDelegatedAPIs();
+
+    //Replace this pls?
+    await apiDelegations.delegateAPI(standardApiDetails.name, standardOrgUser.orgNumber);
+
+    await expect(apiDelegations.giveAccessToNewApiHeading).toBeVisible();
+
+    await expect(page.getByText(standardApiDetails.name)).toBeVisible();
+    await expect(page.getByText(standardApiDetails.department)).toBeVisible();
+    await expect(page.getByText(`Org.nr. ${standardOrgUser.orgNumber}`)).toBeVisible();
+
+    await page.getByRole('button', { name: 'Bekreft' }).click();
+
+    await expect(
+      page.getByRole('heading', { name: 'Disse api-delegeringene ble gitt' }),
+    ).toBeVisible();
+
+    await expect(page.getByText(standardOrgUser.reportee)).toBeVisible();
+    await expect(page.getByText(standardApiDetails.name)).toBeVisible();
+  });
+});
+
+test('Verify adding multiple organizations and APIs and deleting them', async ({
+  login,
+  apiDelegations,
+  page,
+}) => {
+  //Login and cleanup state before running test
+  await login.loginWithUser(standardDaglUser.id);
+  await login.chooseReportee(standardDaglUser.reportee);
+
+  await apiDelegations.deleteDelegatedAPIs();
+
+  await apiDelegations.goToAccessToApiPageFromFrontPage();
+
+  await apiDelegations.delegateNewApiButton.click();
+  await apiDelegations.filterByAgencyButton.click();
+
+  await page.getByLabel(standardApiDetails.department).click();
+  await apiDelegations.applyButton.click();
+
+  const apiNames = [
+    'AuthorizedParties: Mainunit to Organization Maskinporten Schema for utviklingstester',
+    'Automation Regression',
+    'AuthorizedParties: Subunit to Organization Maskinporten Schema for utviklingstester',
+  ];
+
+  await apiDelegations.selectApiAccess(apiNames);
+
+  //  Select users that gets granted access
+  const organizations: [string, string][] = [
+    ['310661414', 'INTERESSANT KOMPATIBEL TIGER AS'],
+    ['310110914', 'LYKKELIG DRIFTIG APE'],
+    ['313259412', 'UROMANTISK ALTERNATIV KATT GÅS'],
+  ];
+
+  await apiDelegations.grantUserAccess(organizations);
+  await apiDelegations.confirmAccessGranted(apiNames, organizations);
+  await apiDelegations.verifyDelegatedRightsGiven(apiNames, organizations);
+});
+
+test('Verify that Tilgangsstyrer does NOT have access to API delegering panel', async ({
+  page,
+  login,
+  apiDelegations,
+}) => {
+  const testUser = '14824497789';
+  const reporteeWithoutAccess = 'BLÅVEIS SKRAVLETE';
```

Comment:
Reportee should be org user as API-delegering should panel available for org user or if you want to verify without keyrole(Tilgangstyrring or HADM) if user gets API-delegering panel can be verified with UTINN roll(altinn roll)